packages/flutter_tools/test/general.shard/ios/xcodeproj_test.dart

```diff
@@ -1110,196 +1111,119 @@ Information about project "Runner":
-        'excludes arm64 simulator when build setting fetch fails',
-        () async {
-          const BuildInfo buildInfo = BuildInfo.debug;
-          final FlutterProject project = FlutterProject.fromDirectoryTest(
-            fs.directory('path/to/project'),
-          );
-          final Directory podXcodeProject =
-              project.ios.hostAppRoot.childDirectory('Pods').childDirectory('Pods.xcodeproj')
-                ..createSync(recursive: true);
-
-          final String buildDirectory = fileSystem.path.absolute('build', 'ios');
-          fakeProcessManager.addCommands(<FakeCommand>[
-            kWhichSysctlCommand,
-            kARMCheckCommand,
-            FakeCommand(
-              command: <String>[
-                '/usr/bin/arch',
-                '-arm64e',
-                'xcrun',
-                'xcodebuild',
-                '-alltargets',
-                '-sdk',
-                'iphonesimulator',
-                '-project',
-                podXcodeProject.path,
-                '-showBuildSettings',
-                'BUILD_DIR=$buildDirectory',
-                'OBJROOT=$buildDirectory',
-              ],
-              exitCode: 1,
-            ),
-          ]);
-          await updateGeneratedXcodeProperties(project: project, buildInfo: buildInfo);
+        const rubyScript = r'''
+              class MockTarget
+                def platform_name; :ios; end
+                def name; 'MyPlugin'; end
+                def deployment_target; '13.0'; end
+              end
+              require ARGV[0]
+              target = MockTarget.new
+              flutter_additional_ios_build_settings(target)
+              ''';
+
+        fakeProcessManager.addCommand(
+          FakeCommand(
+            command: <String>['ruby', '-e', rubyScript, podhelperPath],
+            workingDirectory: fs.path.join('ios'),
+            exitCode: 1,
+            stderr:
+                'The pod `MyPlugin` config in ${xcfile.path} excludes the arm64 simulator architecture.',
+          ),
+        );
 
-          final File config = fs.file('path/to/project/ios/Flutter/Generated.xcconfig');
-          expect(
-            config.readAsStringSync(),
-            contains('EXCLUDED_ARCHS[sdk=iphonesimulator*]=i386 arm64\n'),
-          );
-          expect(config.readAsStringSync(), contains('EXCLUDED_ARCHS[sdk=iphoneos*]=armv7\n'));
-          expect(fakeProcessManager, hasNoRemainingExpectations);
-        },
-        overrides: <Type, Generator>{
-          Artifacts: () => localIosArtifacts,
-          Platform: () => macOS,
-          FileSystem: () => fs,
-          ProcessManager: () => fakeProcessManager,
-          XcodeProjectInterpreter: () => xcodeProjectInterpreter,
-        },
-      );
+        final ProcessResult result = await fakeProcessManager.run(<String>[
+          'ruby',
+          '-e',
+          rubyScript,
+          podhelperPath,
+        ], workingDirectory: fs.path.join('ios'));
```

Comment:
This should be made into an integration test instead of a unit test. This test doesn't really verify your fix. Since you're using a `fakeProcessManager`, it doesn't actually run the `ruby` command or anything, it just returns the `stderr` from the `FakeCommand`.

See https://github.com/flutter/flutter/blob/master/packages/flutter_tools/test/integration.shard/swift_package_manager_test.dart as an example